From 8d1b3d71fdcdaa93c1ae248c443b68170c2d1856 Mon Sep 17 00:00:00 2001
From: John Syn <john3909@gmail.com>
Date: Tue, 15 Jul 2014 19:14:24 -0500
Subject: [PATCH 2/2] cape: audio

I have attached two files that enable Audio Cape support in the V3.15
kernel. You may want to add these to your Audio Cape wiki. In the
am335x-bone-common.dtsi file, I added the tlv320aic3x codec definition to
i2c2, which also includes the regulator definitions which fix the
regulator errors when the codec is loaded. The
am335x-boneblack-cape-audi.dts is just an augmented version of
am335x-boneblack.dts where I added the pinmux for McASP and added the
mcasp and sound definitions.

Signed-off-by: John Syn <john3909@gmail.com>
Signed-off-by: Robert Nelson <robertcnelson@gmail.com>
---
 arch/arm/boot/dts/am335x-bone-audi.dts      | 46 +++++++++++++++++++++++++++
 arch/arm/boot/dts/am335x-boneblack-audi.dts | 49 +++++++++++++++++++++++++++++
 2 files changed, 95 insertions(+)
 create mode 100644 arch/arm/boot/dts/am335x-bone-audi.dts
 create mode 100644 arch/arm/boot/dts/am335x-boneblack-audi.dts

diff --git a/arch/arm/boot/dts/am335x-bone-audi.dts b/arch/arm/boot/dts/am335x-bone-audi.dts
new file mode 100644
index 0000000..cff860b
--- /dev/null
+++ b/arch/arm/boot/dts/am335x-bone-audi.dts
@@ -0,0 +1,46 @@
+/*
+ * Copyright (C) 2012 Texas Instruments Incorporated - http://www.ti.com/
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+/dts-v1/;
+
+#include "am33xx.dtsi"
+#include "am335x-bone-common.dtsi"
+#include "am335x-bone-common-pinmux.dtsi"
+
+&am33xx_pinmux {
+	pinctrl-names = "default";
+	pinctrl-0 = <&clkout2_pin>;
+};
+
+&ldo3_reg {
+	regulator-min-microvolt = <1800000>;
+	regulator-max-microvolt = <3300000>;
+	regulator-always-on;
+};
+
+&mmc1 {
+	vmmc-supply = <&ldo3_reg>;
+};
+
+&sham {
+	status = "okay";
+};
+
+&aes {
+	status = "okay";
+};
+
+/* spiX */
+/* #include "am335x-bone-spi0.dtsi" */
+
+/* uartX */
+/* #include "am335x-bone-ttyO1.dtsi" */
+/* #include "am335x-bone-ttyO2.dtsi" */
+/* #include "am335x-bone-ttyO4.dtsi" */
+/* #include "am335x-bone-ttyO5.dtsi" */
+
+#include "am335x-bone-audi.dtsi"
diff --git a/arch/arm/boot/dts/am335x-boneblack-audi.dts b/arch/arm/boot/dts/am335x-boneblack-audi.dts
new file mode 100644
index 0000000..c7fed7a1
--- /dev/null
+++ b/arch/arm/boot/dts/am335x-boneblack-audi.dts
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) 2012 Texas Instruments Incorporated - http://www.ti.com/
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ */
+/dts-v1/;
+
+#include "am33xx.dtsi"
+#include "am335x-bone-common.dtsi"
+#include "am335x-bone-common-pinmux.dtsi"
+
+&ldo3_reg {
+	regulator-min-microvolt = <1800000>;
+	regulator-max-microvolt = <1800000>;
+	regulator-always-on;
+};
+
+&mmc1 {
+	vmmc-supply = <&vmmcsd_fixed>;
+};
+
+/* spiX */
+/* #include "am335x-bone-spi0.dtsi" */
+
+/* uartX */
+/* #include "am335x-bone-ttyO1.dtsi" */
+/* #include "am335x-bone-ttyO2.dtsi" */
+/* #include "am335x-bone-ttyO4.dtsi" */
+/* am335x-boneblack: hdmi has to be disabled for ttyO5 */
+/* #include "am335x-bone-ttyO5.dtsi" */
+
+/* Max Core Speed */
+#include "am335x-boneblack-1ghz.dtsi"
+/* #include "am335x-boneblack-800mhz.dtsi" */
+/* #include "am335x-boneblack-600mhz.dtsi" */
+/* #include "am335x-boneblack-300mhz.dtsi" */
+
+#include "am335x-boneblack-emmc.dtsi"
+
+&am33xx_pinmux {
+	pinctrl-names = "default";
+	pinctrl-0 = <&clkout2_pin>;
+};
+
+#include "am335x-boneblack-nxp-hdmi.dtsi"
+
+#include "am335x-bone-audi.dtsi"
-- 
2.0.1

